openapi/components/schemas/CouponRestrictionRestrictToCustomerTags.yaml

```
@@ -0,0 +1,25 @@
+type: object
+description: Restricts a coupon to customers with specific tags.
+required:
+  - type
+  - tags
+  - requireAllTags
+properties:
+  type:
+    type: string
+    description: Type of coupon restriction.
+    enum: ["restrict-to-customer-tags"]
+  tags:
+    type: array
+    description: Customer tags on which a coupon can be applied.
+    items:
+      type: string
+  requireAllTags:
+    type: boolean
+    description: Determines the behavior for checking customer tags.
+    enum:
+      - true
+      - false
+    x-enumDescriptions:
+      true: Customer must have all listed tags to redeem a coupon.
+      false: Customer must have at least one of the listed tags to redeem a coupon.
```

nitpick: This could be a real enum named something like this:
```suggestion
  mode:
    type: string
    description: Determines the behavior for checking customer tags.
    enum:
      - all
      - any
    x-enumDescriptions:
      all: Customer must have all listed tags to redeem a coupon.
      any: Customer must have at least one of the listed tags to redeem a coupon.
```